pyrelational/samplers/abstract_sampler.py

```diff
@@ -0,0 +1,43 @@
+"""Collection of samplers for active learning strategies."""
+
+from abc import ABC, abstractmethod
+from typing import List
+
+import torch
+from torch import Tensor
+
+
+class AbstractSampler(ABC):
+    """Abstract sampler class."""
+
+    @abstractmethod
+    def __call__(self, scores: Tensor, indices: List[int], num_samples: int) -> List[int]:
+        """Sample a subset of indices based on the scores.
+
+        This method should be implemented by the subclasses.
+        :return: List of sampled indices.
+        """
+        pass
+
+
+class DeterministicSampler(AbstractSampler):
```

Comment:
Again just a discussion on naming. I think it would be good to just call it a `TopKSampler`. There are other deterministic batch samplers out there.

Reply:
This would kind of imply there would be a K hyperparameter that is fixed

Reply:
The `num_samples` would be this K. I think the naming would be ok in the same way it is used in https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.pool.TopKPooling.html

Reply:
it's not fixed though, as we pass it to __call__. I think Deterministic makes more sense if we call the other Probabilistic